lib/src/kbasesearchengine/parse/KeywordParser.java

```diff
@@ -254,31 +254,27 @@ private static Object transform(Object value, IndexingRules rule,
         case guid:
             String type = transform.getTargetObjectType().get();
             if (type == null) {
-                throw new IllegalStateException("Target object type should be set for 'guid' " +
+                throw new ObjectParseException("Target object type should be set for 'guid' " +
                         "transform");
             }
             final ObjectTypeParsingRules typeDescr = lookup.getTypeDescriptor(type);
             final String storageCode = typeDescr.getStorageObjectType().getStorageCode();
             final Set<String> refs = toStringSet(value);
             final Set<GUID> unresolvedGUIDs;
-            try {
-                unresolvedGUIDs = refs.stream().map(r -> GUID.fromRef(storageCode, r))
+            unresolvedGUIDs = refs.stream().map(r -> GUID.fromRef(storageCode, r))
                         .collect(Collectors.toSet());
-            } catch (IllegalArgumentException e) {
-                throw new ObjectParseException(e.getMessage(), e);
-            }
```

Comment:
in this case the illegal argument exception is thrown because the reference extracted from the input object is illegal, so an objectparseexception is the right exception to throw.

Reply:
Reverted